config/routes/api.rb

```diff
@@ -1,80 +1,84 @@
-      member do
-        put :capture
-        put :ship
+        resources :variants
       end
 
-      resources :shipments, :only => [:create, :update] do
+      resources :variants, :only => [:index]
+
+      resources :orders, only: [:index, :show] do
         member do
-          put :ready
+          put :capture
           put :ship
-          put :add
-          put :remove
+        end
+
+        resources :shipments, :only => [:create, :update] do
+          member do
+            put :ready
+            put :ship
+            put :add
+            put :remove
+          end
         end
       end
-    end
 
-    resources :enterprises do
-      post :update_image, on: :member
+      resources :enterprises do
+        post :update_image, on: :member
 
-      resource :logo, only: [:destroy]
-      resource :promo_image, only: [:destroy]
-      resource :terms_and_conditions, only: [:destroy]
-    end
+        resource :logo, only: [:destroy]
+        resource :promo_image, only: [:destroy]
+        resource :terms_and_conditions, only: [:destroy]
+      end
 
-    resources :shops, only: [:show] do
-      collection do
-        get :closed_shops
+      resources :shops, only: [:show] do
+        collection do
+          get :closed_shops
+        end
       end
-    end
 
-    resources :order_cycles do
-      get :products, on: :member
-      get :taxons, on: :member
-      get :properties, on: :member
-    end
+      resources :order_cycles do
+        get :products, on: :member
+        get :taxons, on: :member
+        get :properties, on: :member
+      end
 
-    resources :exchanges, only: [:show], to: 'exchange_products#index' do
-      get :products, to: 'exchange_products#index'
-    end
+      resources :exchanges, only: [:show], to: 'exchange_products#index' do
+        get :products, to: 'exchange_products#index'
+      end
 
-    resource :status do
-      get :job_queue
-    end
+      resource :status do
+        get :job_queue
+      end
 
-    resources :customers, only: [:index, :update]
+      resources :customers, only: [:index, :update]
 
-    resources :enterprise_fees, only: [:destroy]
+      resources :enterprise_fees, only: [:destroy]
 
-    post '/product_images/:product_id', to: 'product_images#update_product_image'
+      post '/product_images/:product_id', to: 'product_images#update_product_image'
 
-    resources :states, :only => [:index, :show]
+      resources :states, :only => [:index, :show]
 
-    resources :taxons, :only => [:index]
+      resources :taxons, :only => [:index]
 
-    resources :taxonomies do
-      member do
-        get :jstree
-      end
-      resources :taxons do
+      resources :taxonomies do
         member do
           get :jstree
         end
+        resources :taxons do
+          member do
+            get :jstree
+          end
+        end
       end
     end
+
+    match '*path', to: redirect("/api/v0/%{path}"), via: :all, constraints: { path: /(?!.*v0).+/ }
```

Comment:
This is the only bit that concerns me. Can we add a couple of simple tests for the routing? I imagine we'll want to change this at some point when we start adding add the `v1` routes.

Reply:
I think we can get rid of the `.*` in the regex, and it'll only match the `v0` part at the start of the path and not the whole thing. I wonder if we need to check how `GET` params are handled when passed in the URL as well?

Reply:
> I wonder if we need to check how GET params are handled when passed in the URL as well?

I checked, and they were being ignored. Fixed now: fc5909a8a8ae4a6f37515702cf48e2c5efd49f75

> I think we can get rid of the .* in the regex, and it'll only match the v0 part at the start of the path and not the whole thing

Yep, agree: 33d15d2c886fb229e36e5aecb342aa81e536b5dc

> I imagine we'll want to change this at some point...

Might as well do it now I think: bd112515acecf53e685a5bc7817995e7a31c6ef5

> Can we add a couple of simple tests for the routing?

Definitely agree. Coming soon.

Reply:
Here's the first two I thought of: 4052e2acc2ca077caeeb429f117ae4e407c10f1c I might think of some more, but in writing those I noticed that we're going to have to think about authentication pretty soon, if we haven't already. But I think that's outside the scope of this PR.

Reply:
Great :muscle: